### PR TITLE
cleaning up calls to lists:foldl

### PR DESCRIPTION
**Pull request by shamouda**
Reviewing calls to lists:foldl and optimizing them when possible (e.g. by replacing foldl with a list comprehension or a more efficient list function)

### CODE REVIEW

File: src/clocksi_vnode.erl [hunk 1]
```diff
@@ -150,15 +150,7 @@ get_active_txns_internal(TableName) ->
...
-                        lists:foldl(fun({_Key, List}, Acc) ->
-                            case List of
-                                [] ->
-                                    Acc;
-                                _ ->
-                                    List ++ Acc
-                            end
-                        end,
-                            [], [{Key1, List1} | Rest1])
+                        lists:flatmap(fun({_, _List}) -> _List end, [{Key1, List1} | Rest1])
```
Comment (bieniusa) [inline src/clocksi_vnode.erl]: Why `_List`? Couldn't it be simply `List`?

File: src/clocksi_vnode.erl [hunk 2]
```diff
@@ -150,15 +150,7 @@ get_active_txns_internal(TableName) ->
                     [] ->
```
Comment (albsch) [inline src/clocksi_vnode.erl]: This case check can be left out completely, no?
`lists:flatmap` also works on the empty list `[]`, and we do not do anything special if the list has at least one element.

File: src/clocksi_vnode.erl (hunk 1 above)
Comment (shamouda) [reply src/clocksi_vnode.erl]: Just committed a fix: https://github.com/AntidoteDB/antidote/pull/414/commits/cb6ce9fc54e26929a16250b8fc33a487e5905154

File: src/clocksi_vnode.erl (hunk 2 above)
Comment (shamouda) [reply src/clocksi_vnode.erl]: Just committed a fix: https://github.com/AntidoteDB/antidote/pull/414/commits/cb6ce9fc54e26929a16250b8fc33a487e5905154

**Pull request merged**